pure_pursuit_wrapper/src/pure_pursuit_wrapper.cpp

```diff
@@ -86,34 +229,37 @@ void PurePursuitWrapper::trajectoryPlanHandler(const cav_msgs::TrajectoryPlan::C
     }
   }
 
-  std::vector<double> lag_speeds = apply_response_lag(speeds, downtracks, config_.vehicle_response_lag); // This call requires that the first speed point be current speed to work as expected
-
-  autoware_msgs::Lane lane;
-  lane.header = tp->header;
-  std::vector<autoware_msgs::Waypoint> waypoints;
-  waypoints.reserve(trajectory_points.size());
-
-  for (int i = 0; i < trajectory_points.size(); i++)
+  std::vector<double> lag_speeds;
+  if (config_.is_delay_compensation)
+    lag_speeds = apply_response_lag(speeds, downtracks, config_.vehicle_response_lag); // This call requires that the first speed point be current speed to work as expected
+  else
+    lag_speeds = speeds;
+    
+  autoware_auto_msgs::msg::Trajectory autoware_trajectory;
+  autoware_trajectory.header = tp.header;
+  RCLCPP_ERROR_STREAM(rclcpp::get_logger("pure_pursuit_wrapper"), "size: " << trajectory_points.size());
+  auto max_size = std::min(99, (int)trajectory_points.size());
```

Comment:
Is 99 and autoware auto limit?

Reply:
Yeah this was autoware limit, 100 was working, but more than it was failing. So I capped it here, and I think it is safe to do so as I don't think we need that many points.

Reply:
added a comment to explain